### PR TITLE
Autogenerate customer, disputes, invoices and search resources

### PR DESCRIPTION
**Pull request by agualis**

### CODE REVIEW

File: src/resources/disputes-resource.js [hunk 1]
```diff
@@ -1,47 +1,55 @@
...
+      expand = null,
+    } = {}) {
+      const params = {filter, sort, limit, offset, q, expand};
+      return apiHandler.getAll(`disputes`, params);
+    },
+    create({id = '', data, expand = null}) {
+      const params = {expand};
+      return apiHandler.create(`disputes/${id}`, id, data, params);
+    },
+    get({id}) {
```
Comment (cesarlevel) [inline src/resources/disputes-resource.js:26]: There is an `expand` missing

File: src/resources/invoices-resource.js [hunk 2]
```diff
@@ -1,136 +1,126 @@
...
+      expand = null,
+    } = {}) {
+      const params = {filter, sort, limit, offset, q, expand};
+      return apiHandler.getAll(`invoices`, params);
+    },
+    create({id = '', data, expand = null}) {
+      const params = {expand};
+      return apiHandler.create(`invoices/${id}`, id, data, params);
+    },
+    get({id, Accept, expand = null}) {
```
Comment (cesarlevel) [inline src/resources/invoices-resource.js:26]: What is `Accept` here?

File: src/resources/invoices-resource.js [hunk 3]
```diff
@@ -1,136 +1,126 @@
...
+    issue({id, data}) {
+      return apiHandler.post(`invoices/${id}/issue`, data);
+    },
+    abandon({id}) {
+      return apiHandler.post(`invoices/${id}/abandon`);
+    },
+    void({id}) {
+      return apiHandler.post(`invoices/${id}/void`);
+    },
+    recalculate({id}) {
```
Comment (cesarlevel) [inline src/resources/invoices-resource.js:55]: This is a new endpoint :D

File: src/resources/invoices-resource.js [hunk 4]
```diff
@@ -1,136 +1,126 @@
...
+      return apiHandler.post(`invoices/${id}/recalculate`);
+    },
+    reissue({id, data}) {
```
Comment (cesarlevel) [inline src/resources/invoices-resource.js:58]: This one too :)

File: src/resources/invoices-resource.js [hunk 5]
```diff
@@ -1,136 +1,126 @@
...
+      offset = null,
+      filter = null,
+      sort = null,
+      q = null,
+      expand = null,
+    } = {}) {
+      const params = {limit, offset, filter, sort, q, expand};
+      return apiHandler.getAll(`invoices/${id}/timeline`, params);
+    },
+    createTimelineComment({id, data}) {
```
Comment (cesarlevel) [inline src/resources/invoices-resource.js:88]: On the "old" one there is an empty string `return apiHandler.create(`invoices/${id}/timeline`, '', data)` not sure what that means, maybe an error on the handmade one?

File: src/resources/invoices-resource.js (hunk 2 above)
Comment (agualis) [reply src/resources/invoices-resource.js]: Good catch. There was a bug in the generator, that was including `header` parameters.

File: src/resources/invoices-resource.js (hunk 5 above)
Comment (agualis) [reply src/resources/invoices-resource.js:85]: Yeah, that was working but it was odd. 
Explanation: 
`apiHandler.create` calls `put` or `post` depending if you pass an `id` or not
We were always passing an empty `id` with means that can replace the old `apiHandler.create` with a `apiHandler.post` that does not need `id` parameter.

In this case: 🤖 won 🧑‍💻

File: src/resources/disputes-resource.js (hunk 1 above)
Comment (agualis) [reply src/resources/disputes-resource.js:26]: I will ask about this in `pd-mentoring.`

**Review by cesarlevel — DISMISSED** [review]

**Review by ashkarpetin — DISMISSED** [review]

**Review by ashkarpetin — APPROVED** [review]

**Pull request merged**